include/ceed-impl.h

```diff
@@ -66,6 +66,7 @@ struct CeedVector_private {
   int (*RestoreArrayRead)(CeedVector, const CeedScalar **);
   int (*Destroy)(CeedVector);
   int refcount;
+  int state;
```

Comment:
Signed overflow is undefined and can happen with a 32-bit integer (a billion time steps is a lot, but not impossible). I think `uint64_t` would be preferable here.